### DISCUSSION TITLE
Error npm install

### DISCUSSION THREAD

**Discussion by Andrea-Graf** [web-classroom/web-classroom.github.io#42] [Q&A : Pratique / Labos] [answered]
andrea@DESKTOP-J4894UA:/mnt/c/Users/Andrea Graf/OneDrive - HESSO/Documents/HEIG-VD/SEM4/WEB/1-html-css-Andrea-Graf$ npm install

changed 1 package, and audited 2437 packages in 30s

171 packages are looking for funding
  run `npm fund` for details

55 vulnerabilities (1 low, 27 moderate, 26 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
andrea@DESKTOP-J4894UA:/mnt/c/Users/Andrea Graf/OneDrive - HESSO/Documents/HEIG-VD/SEM4/WEB/1-html-css-Andrea-Graf$ npm run 
dev

> 01-html-css@1.0.0 dev
> live-server ./public

Serving "./public" at http://127.0.0.1:8080
/mnt/c/Users/Andrea Graf/OneDrive - HESSO/Documents/HEIG-VD/SEM4/WEB/1-html-css-Andrea-Graf/node_modules/opn/index.js:85
                                        reject(new Error('Exited with code ' + code));
                                               ^

Error: Exited with code 3
    at ChildProcess.<anonymous> (/mnt/c/Users/Andrea Graf/OneDrive - HESSO/Documents/HEIG-VD/SEM4/WEB/1-html-css-Andrea-Graf/node_modules/opn/index.js:85:13)
    at Object.onceWrapper (node:events:632:26)
    at ChildProcess.emit (node:events:517:28)
    at maybeClose (node:internal/child_process:1098:16)
    at Socket.<anonymous> (node:internal/child_process:450:11)
    at Socket.emit (node:events:517:28)
    at Pipe.<anonymous> (node:net:350:12)

Node.js v18.19.1

**Comment by Nortalle** [comment, marked as answer]
Hi !

So on WSL you need to modify some scripts in your `package.json` in order for the live server to work:

```json
    "dev": "live-server ./public --no-browser",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run",
    "test": "cypress run",
```

and always on WSL, if cypress does not work, install the dependencies for cypress [dependencies](https://docs.cypress.io/guides/continuous-integration/introduction#UbuntuDebian):

```bash
apt-get install libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libnss3 libxss1 libasound2 libxtst6 xauth xvfb
```
